### PR TITLE
searchapp: move reusable functions

### PR DESCRIPTION
**Pull request by jrcastro2**
* move reusable functions to
  invenio-search-ui
* requires https://github.com/inveniosoftware/invenio-search-ui/pull/131
Co-authored-by: Pablo pineirin@gmail.com

### CODE REVIEW

File: invenio_app_rdm/records_ui/searchapp.py [hunk 1]
```diff
@@ -7,250 +7,46 @@
...
     return {
-        'search_app_config': partial(
+        'search_app_rdm_config': partial(
             search_app_config,
             'RDM_SEARCH',
             current_app.config['RDM_FACETS'],
-            '/api/records'),
-        'search_app_user_uploads_config': partial(
+            current_app.config['RDM_SORT_OPTIONS'],
+            '/api/records',
```
Comment (kpsherva) [inline invenio_app_rdm/records_ui/searchapp.py:24]: I am not really a fan of hardcoded URLs, can we get it from any config?

File: invenio_app_rdm/records_ui/searchapp.py [hunk 2]
```diff
@@ -233,24 +20,33 @@ def search_app_context():
             search_app_config,
             'RDM_SEARCH',
             current_app.config['RDM_FACETS'],
-            '/api/records'),
+            current_app.config['RDM_SORT_OPTIONS'],
+            '/api/records',
+            {"Accept": "application/vnd.inveniordm.v1+json"}
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_app_rdm/records_ui/searchapp.py:25]: sorry, silly question. Why in some searches we have
```
{"Accept": "application/vnd.inveniordm.v1+json"}
```
and in the others

```
{"Accept": "application/json"}
```

?

File: invenio_app_rdm/records_ui/searchapp.py [hunk 3]
```diff
@@ -7,250 +7,46 @@
...
-        'search_app_user_requests_config': partial(
+            current_app.config['RDM_SORT_OPTIONS'],
+            '/api/user/communities',
+            {"Accept": "application/json"}
+        ),
+        'search_app_rdm_user_requests_config': partial(
             search_app_config,
             'RDM_SEARCH_USER_REQUESTS',
             current_app.config['REQUESTS_FACETS'],
+            current_app.config['RDM_SORT_OPTIONS'],
```
Comment (kpsherva) [inline invenio_app_rdm/records_ui/searchapp.py:47]: I understand there is no sort options for the user requests and we must use the global ones, is that correct?

File: invenio_app_rdm/records_ui/searchapp.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ntarocco) [reply invenio_app_rdm/records_ui/searchapp.py:25]: @kpsherva https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/config.py#L40

File: setup.py [hunk 4]
```diff
@@ -68,6 +68,7 @@
     f'invenio[base,auth,metadata,files]{invenio_version}',
     'invenio-logging[sentry-sdk]>=1.3.0,<1.4.0',
     'invenio-rdm-records>=0.35.1,<0.36.0',
+    'invenio-search-ui>=1.2.0,<1.3.0',
```
Comment (ntarocco) [inline setup.py:71]: it is already part of `invenio` above, so it should not be needed. Can you please check?

Comment (jrcastro2) [reply setup.py:71]: You are right not needed anymore as it is part of `invenio`

File: invenio_app_rdm/records_ui/searchapp.py (hunk 3 above)
Comment (jrcastro2) [reply invenio_app_rdm/records_ui/searchapp.py:47]: Not yet (not sure if there will be), `RDM_SORT_OPTIONS` are the default ones for rdm.

File: invenio_app_rdm/records_ui/searchapp.py (hunk 1 above)
Comment (jrcastro2) [reply invenio_app_rdm/records_ui/searchapp.py:24]: I agree, it doesn't look like we can get all these URLs from an existent config (unless I am mistaken), might be worth to create a new one, any suggestion for the name? maybe `RDM_SEARCH_CONFIG_URLS`

Comment (ntarocco) [reply invenio_app_rdm/records_ui/searchapp.py:24]: As discussed, next PR ;)

**Review by zzacharo — APPROVED** [review]

**Pull request merged**